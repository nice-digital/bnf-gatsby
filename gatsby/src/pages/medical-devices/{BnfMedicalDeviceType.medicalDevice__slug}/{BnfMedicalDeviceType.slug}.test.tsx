import { render, waitFor, screen, within } from "@testing-library/react";

import MedicalDeviceTypePage, {
	MedicalDeviceTypePageProps,
} from "./{BnfMedicalDeviceType.slug}";

const props: MedicalDeviceTypePageProps = {
	data: {
		bnfMedicalDeviceType: {
			title: "Film gloves",
			slug: "film-gloves",
			medicalDevice: {
				title: "Gloves",
				slug: "gloves",
				medicalDeviceTypes: [
					{
						title: "Nitrile gloves",
						slug: "nitrile-gloves",
					},
					{
						title: "Film gloves",
						slug: "film-gloves",
					},
					{
						title: "Polythene gloves",
						slug: "polythene-gloves",
					},
				],
			},
			preparations: [
				{
					name: "Dispos-A-Gloves non-sterile large",
					manufacturer: "Ansell Medical Ltd",
					activeIngredients: [],
					ampId: "123",
					blackTriangle: false,
					controlledDrugSchedule: null,
					packs: [],
					sugarFree: false,
				},
				{
					name: "Dispos-A-Gloves non-sterile medium",
					manufacturer: "Ansell Medical Ltd",
					activeIngredients: [],
					ampId: "123",
					blackTriangle: false,
					controlledDrugSchedule: null,
					packs: [],
					sugarFree: false,
				},
			],
		},
	},
};

describe("MedicalDeviceTypePage", () => {
	describe("SEO", () => {
		it("should set page title from medical type title", async () => {
			render(<MedicalDeviceTypePage {...props} />);
			await waitFor(() => {
				expect(document.title).toStartWith(
					"Film gloves | Gloves | Medical devices | "
				);
			});
		});

		it("should set meta description", async () => {
			render(<MedicalDeviceTypePage {...props} />);
			await waitFor(() => {
				expect(
					// eslint-disable-next-line testing-library/no-node-access
					document.querySelector("meta[name=description]")
				).toHaveAttribute(
					"content",
					"This medical device type describes the options that are currently recommended for Film gloves."
				);
			});
		});
	});

	describe("Breadcrumbs", () => {
		it.each([
			["NICE", "https://www.nice.org.uk/"],
			["BNF", "/"],
			["Medical devices", "/medical-devices/"],
			["Gloves", "/medical-devices/gloves/"],
		])(
			"should render default '(%s)' breadcrumb",
			(breadcrumbText, expectedHref) => {
				render(<MedicalDeviceTypePage {...props} />);

				const breadcrumbNav = screen.getByRole("navigation", {
					name: "Breadcrumbs",
				});
				const breadcrumb = within(breadcrumbNav).getByRole("link", {
					name: breadcrumbText,
				});

				expect(breadcrumb).toHaveAttribute("href", expectedHref);
			}
		);

		it("should render current page breadcrumb without link", () => {
			render(<MedicalDeviceTypePage {...props} />);

			const breadcrumbNav = screen.getByRole("navigation", {
				name: "Breadcrumbs",
			});
			const currentPageCrumb = within(breadcrumbNav).getByText("Film gloves", {
				selector: "span",
			});
			expect(currentPageCrumb).toBeInTheDocument();
		});
	});

	describe("Page header", () => {
		it("should add content start skip link target id to page header", () => {
			render(<MedicalDeviceTypePage {...props} />);
			const heading1 = screen.getByRole("heading", {
				level: 1,
			});
			// eslint-disable-next-line testing-library/no-node-access
			expect(heading1.parentElement).toHaveAttribute("id", "content-start");
		});

		it("should render heading 1 with current page title", () => {
			render(<MedicalDeviceTypePage {...props} />);
			const heading1 = screen.getByRole("heading", {
				level: 1,
			});
			expect(heading1).toHaveTextContent("Film gloves");
		});
	});

	describe("Body", () => {
		describe("Stacked Nav", () => {
			it.todo("should render stacked nav when there are sibling device types");

			it.todo("should render nav link to parent medical device");
			it.todo("should render nav link to each sibling device types");
			it.todo("should render highlight current page");
		});

		it.todo("should render list of preparations");
	});
});
