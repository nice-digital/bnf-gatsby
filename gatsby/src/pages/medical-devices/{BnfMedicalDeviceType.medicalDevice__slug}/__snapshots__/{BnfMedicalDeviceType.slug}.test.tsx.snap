// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicalDeviceTypePage Body Indications and dose should match snapshot for indications and dose section 1`] = `
<section
  aria-labelledby="indications-and-dose"
>
  <h2
    id="indications-and-dose"
  >
    Indications and dose
  </h2>
  <section
    aria-labelledby="irrigation-solutions-indications-and-dose"
  >
    <h3
      class="visually-hidden"
      id="irrigation-solutions-indications-and-dose"
    >
      For Irrigation solutions
    </h3>
    <section
      aria-labelledby="irrigation-solutions-indication-1"
      class="indicationWrapper"
    >
      <h4
        class="indicationHeading"
        id="irrigation-solutions-indication-1"
      >
        <span
          class="indicationText"
        >
          Skin cleansing
        </span>
        <span
          class="visually-hidden"
        >
           for Irrigation solutions
        </span>
      </h4>
      <section
        class="routeOfAdministration"
      >
        <h5
          class="routeOfAdministrationHeading"
        >
          To the skin
        </h5>
        <dl
          class="patientGroups"
        >
          <div
            class="patientGroupDose child"
          >
            <dt
              class="patientGroup"
            >
              Child
            </dt>
            <dd
              class="doseStatement"
            >
              Use for topical irrigation of wounds.
            </dd>
          </div>
          <div
            class="patientGroupDose adult"
          >
            <dt
              class="patientGroup"
            >
              Adult
            </dt>
            <dd
              class="doseStatement"
            >
              Use for topical irrigation of wounds.
            </dd>
          </div>
        </dl>
      </section>
    </section>
  </section>
</section>
`;
